src/main/marketplace/tax-treaties/tax-treaties-service.js

```diff
@@ -58,26 +62,40 @@ export class TaxTreatiesService {
 	async fetchTaxTreatiesFlagtheory() {
 		try {
 			let fetched = await request.get({ url: FLAGTHEORY_TAX_TREATIES_ENDPOINT, json: true });
-			return fetched[0].map(entity =>
-				_.mapKeys(_.omit(entity, 'id'), (value, key) => _.camelCase(key))
-			);
+			let data = [];
+			for (const entity of fetched[0]) {
+				const item = await mapKeysAsync(_.omit(entity, 'id'), (value, key) =>
+					_.camelCase(key)
+				);
+				data.push(item);
+				await setImmediatePromise();
+			}
+			return data;
 		} catch (error) {
 			log.error(`fetchTaxTreatiesFlagtheory: ${error}`);
 			return [];
 		}
 	}
 	start() {
-		this.schedulerService.queueJob(null, TAX_TREATIES_SYNC_JOB);
+		const now = new Date();
+		this.schedulerService.queueJob(
+			null,
+			TAX_TREATIES_SYNC_JOB,
+			now.setMinutes(now.getMinutes() + 1)
+		);
 	}
-
 	loadTaxTreaties() {
 		return TaxTreaties.findAll();
 	}
-	upsert(upsert) {
-		return TaxTreaties.bulkUpsert(upsert);
+	async upsert(upsert) {
+		const chunks = arrayChunks(upsert, 100);
+		for (const c of chunks) {
+			TaxTreaties.bulkUpsert(c);
+			await setImmediatePromise();
+		}
 	}
-	deleteMany(ids) {
-		return TaxTreaties.deleteMany(ids);
+	async deleteMany(ids) {
+		TaxTreaties.deleteMany(ids);
```

Comment:
should have `return` or `await` near `TaxTreaties.deleteMany(ids)`, otherwise it will not wait for promise

Also, it might be beneficial to use array chunks here as well, becouse sometimes it fails on too many ids